### PR TITLE
biogeme: deprecate as of oct 2018

### PR DESCRIPTION
**Pull request by Amar1729**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

According to [`biogeme`'s homepage](http://biogeme.epfl.ch/), the command-line `pythonbiogeme` has been deprecated since late 2018. The new project, `PandasBiogeme`, is purely a pip package and is hosted on PyPI ([release history](https://pypi.org/project/biogeme/#history)).

From the homepage:
```
PythonBiogeme
    Around 2010, a more flexible version was designed for general purpose parametric models. 
    The modeling language was extended, and based on the Python language.
    A series of discrete choice models were precoded for an easy use.
    Also written in GNU C++, it is distributed on the homebrew package manager.
    The distributions can be found here. 

PandasBiogeme
    In 2018, a completely new version of the software was released.
    It was not anymore a standalone executable, but a Python package.
    The package is written in Python, with the exception of the core calculations of the models, that are written in C++ for the sake of efficiency.
    The motivation was to combine the simplicity of the usage (especially for teaching purposes), with the sophistication provided by Python (for research and applications purposes).
    Morever, the management of the data relies on the package Pandas, which has become the workhorse of data scientists.
    Therefore, the name PandasBiogeme has been adopted.
    It is distributed on the Python Package Index repository.
```

Tagging @michelbierlaire - user account that first introduced `biogeme` package. Is it ok to deprecate this from homebrew? Are most user-installs now focused on the PyPI-distributed `PandasBiogeme` rather than the command-line tool, or is `PythonBiogeme` still being used?

https://formulae.brew.sh/formula/biogeme#default

This is one of the formulae remaining before the python3.8 switch - #47274

### CODE REVIEW

**Comment by SMillerDev** [comment]
I'd say we should switch to the pypy package then.

**Comment by michelbierlaire** [comment]
Dear Paul,

Thank you for your message. Yes, it is OK to label it “deprecated”. 
I have no idea if users still use the old version. But they are encouraged to use the new one anyway. 

Best regards,

Michel

**Comment by Amar1729** [comment]
@SMillerDev should this formula switch to the PyPI version, or should it be introduced as a new formula?

**Comment by bayandin** [comment]
> @SMillerDev should this formula switch to the PyPI version, or should it be introduced as a new formula?

Switch it to pypi version, please

**Comment by Amar1729** [comment]
I'm having a bit of trouble converting this to the python, mostly because i can't quite seem to get the `pandas` dependency building properly with cython inside homebrew. Let's see how this works...

**Comment by michelbierlaire** [comment]
Hi Paul,

Can I help in any way?

Best regards,

Michel


On 22 May 2020, at 23:06, Amar Paul <notifications@github.com<mailto:notifications@github.com>> wrote:



I'm having a bit of trouble converting this to the python, mostly because i can't quite seem to get the pandas dependency building properly with cython inside homebrew. Let's see how this works...

**Comment by Amar1729** [comment]
@michelbierlaire thanks for the offer - really I'm just not used to writing python packages as Homebrew formulae. If you're more comfortable with it feel free to update this formula otherwise I'll take another look at it when I can.

**Comment by michelbierlaire** [comment]
@Amar1729 Well, no, I am not sufficiently familiar with Homebrew. I had in mind that, if you need any specific information about the package itself, let me know.

**Comment by Rylan12** [comment]
Is it worth having biogeme available from Homebrew if it's available through pip?

According to the [Python for Formula Authors documentation page](https://docs.brew.sh/Python-for-Formula-Authors):
>Homebrew generally won’t accept libraries that can be installed correctly with `pip install foo`.

As a user, I'm not sure that I see a benefit to having a pip package available through Homebrew. It seems that using `pip install biogeme` is a better option.

That being said, I'm not a maintainer nor do I use biogeme so take my opinion with a grain of salt.

**Comment by Amar1729** [comment]
@Rylan12 i agree, I'm not sure it's worth it to have biogeme provided by homebrew.

**Comment by Rylan12** [comment]
@SMillerDev Is it worth continuing to work on this or should it just be removed?

File: Formula/biogeme.rb [hunk 1]
```diff
@@ -13,12 +13,50 @@ class Biogeme < Formula
...
+    vendor_site_packages = libexec/"vendor/lib/python#{xy}/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", vendor_site_packages
+
+    ENV.prepend_create_path "PYTHONPATH", buildpath/"cython/lib/python#{xy}/site-packages"
+    resource("Cython").stage do
+      system Formula["python@3.8"].opt_bin/"python3", *Language::Python.setup_install_args(buildpath/"cython")
+    end
+    %w[numpy scipy].each do |d|
+      ENV.prepend "PYTHONPATH", Formula[d].opt_lib/"python#{xy}/site-packages"
+    end
```
Comment (Rylan12) [inline Formula/biogeme.rb:47]: Removing these lines did the trick for me

```suggestion
```

**Review by Rylan12 — COMMENTED** [review]
Assuming this is still wanted, this should solve the problems.

The test should also be updated because the current one uses the old binary. Something like this might work:

``` ruby
  test do
    xy = Language::Python.major_minor_version Formula["python@3.8"].opt_bin/"python3"
    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
    system Formula["python@3.8"].opt_bin/"python3", "-c", "import biogeme"
  end
```

**Comment by michelbierlaire** [comment]
Dear all,

Indeed, the previous version of Biogeme was written in C++, using the autoconf/automake environnement. This is why I decided to make it available on Homebrew.
The current version is a pure Python package. Also, I am making the source available on github. https://github.com/michelbierlaire/biogeme

So, it is no problem for me if you decide to remove Biogeme from Homebrew.

Best regards,

Michel


On 29 May 2020, at 18:48, Rylan Polster <notifications@github.com<mailto:notifications@github.com>> wrote:



Is it worth having biogeme available from Homebrew if it's available through pip?

According to the Python for Formula Authors documentation page<https://docs.brew.sh/Python-for-Formula-Authors>:

Homebrew generally won’t accept libraries that can be installed correctly with pip install foo.

As a user, I'm not sure that I see a benefit to having a pip package available through Homebrew. It seems that using pip install biogeme is a better option.

That being said, I'm not a maintainer nor do I use biogeme so take my opinion with a grain of salt.

**Comment by iMichka** [comment]
As upstream is fine with deletion; and it is shipped on pypi, and this is now a pure Python package, and this is the last blocker for the Python 3.8 migration, lets delete this in #55702.

**Pull request closed without merging**